app/src/main/java/waiter/Connectable/ClientConnection.java

```diff
@@ -38,8 +40,36 @@ public void close() throws IOException {
         this.socket.close();
     }
 
-    private boolean notEndOfLine(StringBuilder requestStartLineBuilder) {
-        return requestStartLineBuilder.indexOf("\r\n") == -1;
+    private boolean notEndOfRequest(StringBuilder requestMessageBuilder) {
```

Comment:
Non-blocking: Some of these private functions (here and below) seem like they could live in a class that is more specifically related to parsing requests rather than in the `ClientConnection`.

Reply:
+1

Reply:
Now that you mention it I completely see it and agree, I was so focused on trying to make these named properly and single responsibility when it came to the individual functions that I missed they belong elsewhere, Ill be sure to move them over, thank you for the catch! 😄